powertools-idempotency/src/main/java/software/amazon/lambda/powertools/idempotency/persistence/DynamoDBPersistenceStore.java

```diff
@@ -126,6 +126,19 @@ public DataRecord getRecord(String idempotencyKey) throws IdempotencyItemNotFoun
         return itemToRecord(response.item());
     }
 
+    /**
+     * Store's the given idempotency record in the DDB store. If there
```

Comment:
Trying to make it clearer what this method is doing - both the conditional updating of the existing record and the fact that it doesn't write the _complete_ DataRecord seems important.